laboratoare/open-source/teme/tema-2.markdown

```diff
@@ -0,0 +1,111 @@
+
+Pe lângă sistemul de operare instalat în cadrul acestei mașini virtuale va trebui să adăugați un serviciu care va rula o singură dată, următoarea dată când mașina virtuală va fi pornită.
+Serviciul adăugat va trebui să îndeplinească următorii pași:
+ - să verifice dacă fișierul de configurare (aflat la o locație prestabilită în cadrul sistemului de fișiere) este accesibil -- eventual se poate asigura că locația respectivă devine accesibilă
+ - să citească dintr-un fișier de configurare toate informațiile de care are nevoie pentru a configura sistemul
+ - va seta hostname-ul pentru mașina virtuală
+ - va configura interfața de rețea ce ne permite acces la resurse externe
+ - va instala utilitarul git și va clona depozitul de cod pregătit pentru tema curentă
+ - va lansa în execuție script-ul *boostrap.sh* aflat în depozitul de cod anterior menționat 
+
+### Bootstrap.sh
+
+Pe baza soluției propuse pentru prima temă acasă și pe baza rezolvărilor propuse în cadrul laboratoarelor va trebui să dezvoltați un script bash care se va ocupa de setările inițiale ale sistemului de operare. Script-ul va obține toate informațiile necesare dintr-un fișier de configurare disponibil în cadrul depozitului de cod în care este inclus.
+
+Minimal scriptul va efectua următoarele acțiuni:
+
+- va efectua actualizări pentru programele de sistem existente
+- va configura toate interfețele de rețea disponibilie
+- va configura procesul de SSH astfel încât autentificarea pe bază de parolă nu va mai putea fi efectuată, în schimb, veți putea folosi chei de tip RSA, pe care o veți și instala
+- va reține datele de ieșire și mesajele de eroare în fișierul /var/log/system-bootstrap.log
+- inspectează fișierul /etc/selinux/config și se asigură că parametrul ”SELINUX=” este ”disabled” după care va lansa comanda “setenforce 0”
+
+### VirtualBox Manager
+
+Dezvoltați o aplicație care se va ocupa de gestiunea mașinilor virtuale folosite în cadrul laboratoarelor de *Metodologii de lucru în mediul Open-Source*. Aceasta va trebuie să se folosească de [API-ul expus] de hypervisor-ul [VirtualBox] sau de utilitarul în linie de comandă [VBoxManage].
+
+Aplicația va oferi următoarele funcționalități utilizatorului:
+
+ - `script deploy {hostname}` -- va pregăti o nouă mașină virtuală pe baza mașinii virtuale de bază pregătită anterior
+    - va verifica dacă există un fișier de configurare pentru această mașină virtuală (dacă nu există un fișier de configurare specific va folosi un fișier de configurare prestabilit)
+    - va face o clonă a imaginii de bază
+    - va efectua modifările descrise în cadrul fișierului de configurare (RAM, CPU, discuri, interfețe de rețea etc)
+    - va pregăti fișierul de configurare pentru SystemD unit file și îl va adăuga în mașina virtuală (fie utilizând *shared folder*, fie folosind unitatea optică)
```

Comment:
Pentru ce pregatest acest unit file ? nu ar trebui sa pregatesc fisierul de configurare care va fi digerat de `bootstrap.sh` ? bootstrap.sh poate fi setat ca serviciu de scriptul care face initializarea in imaginea de baza.

Serviciul de Initi Imagine de Baza cloneaza `bootstrap.sh` si il face serviciu si ii da run ...

Imi scapa mie ceva ?

Reply:
În momentul în care va porni mașina virtuală nu va avea conectivitate la rețea. Scriptul care va fi inclus în imaginea de bază (și va fi pornit de către serviciu) va avea nevoie de un set minimal de informații pentru a începe configurarea sistemului, ulterior va obține un script specializat pentru configurarea mașinii virtuale (`boostrap.sh`) pe care îl va executa.

Dacă doriți puteți să adăugați fișierul `bootstrap.sh` direct în imaginea de bază cu mențiunea că după ce mașina va avea conectivitate va trebui să verificați dacă rulați ultima versiune a acestui script.